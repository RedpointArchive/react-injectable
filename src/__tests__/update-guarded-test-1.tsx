import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { InjectGuarded } from '../index';

const FooContext = React.createContext<string | undefined>(undefined);
const BarContext = React.createContext<string | undefined>(undefined);

interface ComponentProps {
  foo: string;
  bar: string;
  baz: string;
  moo: string;
}

const Component = InjectGuarded(
  {
    foo: FooContext,
    bar: BarContext,
  },
  class Component extends React.Component<ComponentProps, {}> {
    public render() {
      return (
        <>
          {this.props.foo}
          {this.props.bar}
          {this.props.baz}
          {this.props.moo}
        </>
      );
    }
  }
);

interface DocumentProps {
  foo: string | undefined;
  bar: string | undefined;
}

class Document extends React.Component<DocumentProps, {}> {
  public render() {
    return (
      <FooContext.Provider value={this.props.foo}>
        <BarContext.Provider value={this.props.bar}>
          <Component baz="baz" moo="moo" />
        </BarContext.Provider>
      </FooContext.Provider>
    );
  }
}

it('nested InjectGuarded component updates correctly', () => {
  const renderer = TestRenderer.create(<Document foo="foo1" bar="bar1" />);

  expect(renderer.toJSON()).toMatchSnapshot();

  renderer.update(<Document foo={undefined} bar="bar1" />);

  expect(renderer.toJSON()).toMatchSnapshot();

  renderer.update(<Document foo={undefined} bar={undefined} />);

  expect(renderer.toJSON()).toMatchSnapshot();

  renderer.update(<Document foo="foo1" bar={undefined} />);

  expect(renderer.toJSON()).toMatchSnapshot();

  renderer.update(<Document foo={undefined} bar="bar1" />);

  expect(renderer.toJSON()).toMatchSnapshot();

  renderer.update(<Document foo="foo2" bar="bar1" />);

  expect(renderer.toJSON()).toMatchSnapshot();
});
