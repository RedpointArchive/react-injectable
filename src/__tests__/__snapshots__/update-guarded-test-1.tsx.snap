// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nested InjectGuarded component updates correctly 1`] = `
Array [
  "foo1",
  "bar1",
  "baz",
  "moo",
]
`;

exports[`nested InjectGuarded component updates correctly 2`] = `null`;

exports[`nested InjectGuarded component updates correctly 3`] = `null`;

exports[`nested InjectGuarded component updates correctly 4`] = `null`;

exports[`nested InjectGuarded component updates correctly 5`] = `null`;

exports[`nested InjectGuarded component updates correctly 6`] = `
Array [
  "foo2",
  "bar1",
  "baz",
  "moo",
]
`;
